components/org.wso2.carbon.editor.log.appender/src/main/java/org/wso2/carbon/editor/log/appender/EditorConsoleAppender.java

```diff
@@ -0,0 +1,132 @@
+     */
+    private EditorConsoleAppender(final String name, final int bufferSize, final Filter filter,
+                                  final Layout<? extends Serializable> layout, final boolean ignoreExceptions) {
+        super(name, filter, layout, ignoreExceptions);
+
+        activateOptions();
+    }
+
+    /**
+     * Taken from the previous EditorConsoleAppender
+     */
+    public void activateOptions() {
+        this.circularBuffer = DataHodlder.getBuffer(BUFFER_SIZE);
+    }
+
+    /**
+     * Creates a EditorConsoleAppender instance with
+     * attributes configured in log4j2.properties.
+     *
+     * @param name   appender name
+     * @param filter null if not specified
+     * @param layout pattern of log messages
+     * @param ignore default is true
+     * @return intance of EditorConsoleAppender
+     */
+    @PluginFactory
+    public static EditorConsoleAppender createAppender(@PluginAttribute("name") final String name,
+                                                       @PluginElement("Filters") final Filter filter,
+                                                       @PluginElement("Layout") Layout<? extends Serializable> layout,
+                                                       @PluginAttribute("ignoreExceptions") final String ignore,
+                                                       @PluginAttribute("buffSize") final String buffSize) {
+        if (name == null) {
+            LOGGER.error("No name provided for EditorConsoleAppender");
+            return null;
+        } else {
+            if (layout == null) {
+                layout = PatternLayout.createDefaultLayout();
+            }
+            final boolean ignoreExceptions = Booleans.parseBoolean(ignore, true);
+            return new EditorConsoleAppender(name, BUFFER_SIZE, filter, layout, ignoreExceptions);
+        }
+    }
+
+    /**
+     * This is the overridden method from the Appender interface. {@link Appender}
+     * This allows to write log events to preferred destination.
+     * <p>
+     * Converts the default log events to tenant aware log events and writes to a CircularBuffer
+     *
+     * @param logEvent the LogEvent object
+     */
+    @Override
+    public void append(LogEvent logEvent) {
+        if (circularBuffer != null) {
+            circularBuffer.append(populateConsoleLogEvent(logEvent));
+        }
+    }
+
+    private ConsoleLogEvent populateConsoleLogEvent(LogEvent logEvent) {
+        ConsoleLogEvent consoleLogEvent = new ConsoleLogEvent();
+        consoleLogEvent.setFqcn(logEvent.getLoggerFqcn());
+        consoleLogEvent.setLevel(logEvent.getLevel().name());
+        consoleLogEvent.setMessage(logEvent.getMessage().getFormattedMessage());
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss_SSS");
+        String dateString = formatter.format(logEvent.getTimeMillis());
+        consoleLogEvent.setTimeStamp(dateString);
+
+        return consoleLogEvent;
+
```

Comment:
Please remove the unnecessary new line.

Reply:
Fixed with [#68464573e9cd97eb8ba4b74b2baa4859a359e17d](https://github.com/wso2/carbon-analytics/commit/68464573e9cd97eb8ba4b74b2baa4859a359e17d)